fix: Switch Button Unification

jjinendra3 opened the pull request:
<!-- This is a pull request template, you do not need to uncomment or remove the comments, they won't show up in the PR text. -->

## Proposed changes
<!-- Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue below. -->
- Added `trackColor` based on the theme.
- Ensured proper visibility in both **light and dark modes**.
- Also added extra support for ios.
## Issue(s)	
<!-- Link the issues being closed by or related to this PR. For example, you can use #594 if this PR closes issue number 594 -->
The avatar switch button was **not visible in light mode** due to inappropriate track color.

## How to test or reproduce
<!-- Mention how you would reproduce the bug if not mentioned on the issue page already. Also mention which screens are going to have the changes if applicable -->
1. Switch to light mode.
2. Go to `Display` screen.
## Screenshots

Before: 

- [Light Mode](https://github.com/user-attachments/assets/9492057a-da9a-40ac-ad56-77daad9d1a34)
- [Dark Mode](https://github.com/user-attachments/assets/d26277ed-1cc8-436a-a595-caa56017dbdf)

After fix:

- [Dark Mode](https://github.com/user-attachments/assets/efb6ea3a-1563-4048-8003-a66ac7f1dd1f)
- [Light Mode](https://github.com/user-attachments/assets/09871309-4541-459e-8306-fa766228c1b8)



## Types of changes
<!-- What types of changes does your code introduce to Rocket.Chat? -->
<!-- Put an `x` in the boxes that apply -->

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist
<!-- Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code. -->

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

diegolmello commented:
#5426

jjinendra3 commented:
Thanks @OtavioStasiak , will implement the colors and update the pr.

jjinendra3 commented:
I have tried to replicate the trackColorFalse to a color similar to Switch default in dark theme.

diegolmello commented:
![image](https://github.com/user-attachments/assets/c702c641-9a78-4be5-8555-502cb74d2aca)

@jjinendra3 These switches look different from the other ones. Can you take a look?

jjinendra3 commented:
Hey @diegolmello , i have made the neccesary changes. Thanks!

jjinendra3 commented:
Hey @diegolmello , just wanted to check in — are the changes looking good to you? Let me know if there's anything you'd like me to tweak or improve. Thanks!

OtavioStasiak commented on app/containers/Switch/index.tsx:
```diff
@@ -7,11 +7,11 @@ const Switch = (props: SwitchProps): React.ReactElement => {
 
 	const trackColor = {
-		false: colors.buttonBackgroundDangerPress,
+		false: colors.fontDisabled,
 		true: colors.buttonBackgroundSuccessPress
```
```suggestion
		false: colors.strokeDark,
		true: colors.buttonBackgroundPrimaryDefault
```
I read your comment about the color, and we can slightly change the order of the colors to get the correct result. I'll add the correct version here.

OtavioStasiak commented on app/containers/Switch/index.tsx:
```diff
@@ -7,11 +7,11 @@ const Switch = (props: SwitchProps): React.ReactElement => {
 	};
 
-	return <RNSwitch trackColor={trackColor} {...props} />;
+	return <RNSwitch trackColor={trackColor} ios_backgroundColor={colors.strokeDark} {...props} />;
```
```suggestion
	return <RNSwitch trackColor={trackColor} thumbColor={colors.fontPureWhite} ios_backgroundColor={colors.strokeDark} {...props} />;
```

OtavioStasiak reviewed (changes requested):
Almost done! I’ve updated the color order. After this small change, we can approve and merge it. Great job!

jjinendra3 commented:
> Almost done! I’ve updated the color order. After this small change, we can approve and merge it. Great job!

Hey @OtavioStasiak, thanks for the heads up! I have updated the file accordingly, thanks very much!

OtavioStasiak reviewed (approved):
Looks good to me, Great job!
Thank you for the contribution.

jjinendra3 commented:
Hey @diegolmello just following up on this PR, please let me know if there’s anything blocking the merge or any changes you'd like me to address!

Happy to take care of anything that’s needed. Thanks!

OtavioStasiak commented:
e2e tests:
https://app.circleci.com/pipelines/github/RocketChat/Rocket.Chat.ReactNative/20430/workflows/b95f89aa-ba16-40bb-b47e-09de44142d63
